src/Nethermind/Nethermind.Facade/BlockchainBridge.cs

```diff
@@ -194,24 +194,25 @@ public CallOutput(byte[] outputData, long gasSpent, string error)
 
         public CallOutput Call(BlockHeader blockHeader, Transaction transaction)
         {
-            CallOutputTracer callOutputTracer = CallAndRestore(blockHeader, transaction);
+            CallOutputTracer callOutputTracer = new CallOutputTracer();
+            CallAndRestore(blockHeader, transaction, callOutputTracer);
             return new CallOutput {Error = callOutputTracer.Error, GasSpent = callOutputTracer.GasSpent, OutputData = callOutputTracer.ReturnValue};
         }
 
         public long EstimateGas(BlockHeader header, Transaction transaction)
         {
-            CallOutputTracer callOutputTracer = CallAndRestore(header, transaction);
-            return callOutputTracer.GasSpent;
+            EstimateGasTracer estimateGasTracer = new EstimateGasTracer();
+            CallAndRestore(header, transaction, estimateGasTracer);
+            return transaction.GasLimit - estimateGasTracer.ExcessiveGas + 1;
```

Comment:
Why +1?  Is there risk of a rounding error?  Gas estimations that have rounding problems can lead to it being difficult to sweep a contract, since you'll always end up leaving dust in the account.

Reply:
thanks, it was the first approach after which I wrote all the required tests and fixed that internally - need to remove it here